### PR TITLE
Swap in tomllib/tomli for toml.

### LINKED ISSUE #339: Switch to tomli/tomllib for toml parsing
Due to https://github.com/uiri/toml/issues/270, vulture won't run for projects with a `pyproject.toml` that contain an array with mixed types. According to https://github.com/python-poetry/poetry/issues/7094#issuecomment-1328127456, this is not an issue in the `tomllib` in Python 3.11+ or its backport (https://github.com/hukkin/tomli).

It looks like this should be a quick swap. I'll send a PR along when I have a moment.

### PR DESCRIPTION
**Pull request by sacsar**
## Description

Fixes #339 -- the toml package does not support heterogenous arrays, which are now allowed in the toml spec. This commit swaps in tomllib from the standard library for Python >= 3.11 and its backport tomli for older versions.

## Checklist:
- [X] I have updated the documentation in the README.md file or my changes don't require an update.
- [x] I have added an entry in CHANGELOG.md.
- [X] I have added or adapted tests to cover my changes.
- [X] I have run `tox -e fix-style` to format my code and checked the result with `tox -e style`.

### CODE REVIEW

**Comment by sacsar** [comment]
Note: While the `tox` style check passes, `flake8` finds errors when run via `pre-commit` (including in `vulture/core.py`, which I didn't touch). I believe this is `flake8` and `black` fighting, but I didn't go too far down the rabbit hole and ended up using `--no-verify`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 2.11 (2023-11-13)
+* Switch to tomllib/tomli to support heterogenous arrays. (Sebastian Csar, #340)
```
Comment (jendrikseipp) [inline CHANGELOG.md:2]:
```suggestion
* Switch to tomllib/tomli to support heterogeneous arrays (Sebastian Csar, #340).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# 2.11 (2023-11-13)
```
Comment (jendrikseipp) [inline CHANGELOG.md:1]:
```suggestion
# next (unreleased)
```

File: tests/test_config.py [hunk 3]
```diff
@@ -75,6 +76,50 @@ def test_toml_config():
...
+        sort_by_size=True,
+        verbose=True,
+    )
+    data = BytesIO(
+        bytes(
+            dedent(
+                """\
+        [tool.foo]
+        # comment for good measure
+        problem_array = [{ a = 1}, [2,3,4], "foo"]
```
Comment (jendrikseipp) [inline tests/test_config.py:109]:
```suggestion
        problem_array = [{a = 1}, [2,3,4], "foo"]
```

File: tests/test_config.py [hunk 4]
```diff
@@ -87,9 +132,10 @@ def test_config_merging():
     If we have both CLI args and a ``pyproject.toml`` file, the CLI args should
     have precedence.
     """
-    toml = StringIO(
-        dedent(
-            """\
+    toml = BytesIO(
```
Comment (jendrikseipp) [inline tests/test_config.py:135]: Could you factor this into a function, please?

```
def get_toml_bytes(config_string):
    return BytesIO(bytes(dedent(config_string), "utf-8"))
```

**Review by jendrikseipp — CHANGES_REQUESTED** [review]
Thanks for the PR! Code looks good, I only have a few minor nitpicks.

File: tests/test_config.py (hunk 4 above)
Comment (sacsar) [reply tests/test_config.py:135]: I'm thinking of going with
```py
def get_toml_bytes(toml_str: str) -> BytesIO:
    """
    Wrap a string in BytesIO to play the role of the incoming config stream.
    """
    return BytesIO(bytes(toml_str, "utf-8"))
```
i.e. keeping `dedent` in the tests themselves. My thinking is that it then becomes clear why one is writing `"""\` rather than `"""`.

**Comment by sacsar** [comment]
> Note: While the `tox` style check passes, `flake8` finds errors when run via `pre-commit` (including in `vulture/core.py`, which I didn't touch). I believe this is `flake8` and `black` fighting, but I didn't go too far down the rabbit hole and ended up using `--no-verify`.

It turns out this isn't a conflict with `black`, it's a change in `flake8` between Python versions. When I run `tox` from a 3.12 virtualenv, the style check fails with:
```
tests/test_utils.py:146:32: E272 multiple spaces before keyword
vulture/config.py:119:69: E231 missing whitespace after ','
vulture/config.py:120:25: E231 missing whitespace after ','
vulture/config.py:120:30: E231 missing whitespace after ','
vulture/config.py:120:42: E231 missing whitespace after ','
vulture/config.py:130:59: E231 missing whitespace after ','
vulture/config.py:138:71: E231 missing whitespace after ','
vulture/core.py:158:41: E231 missing whitespace after ':'
vulture/core.py:172:50: E231 missing whitespace after ':'
```
When run from a 3.8 virtualenv, flake8 is happy. I was getting failures only from pre-commit because I had installed `pre-commit` from the 3.12 virtualenv and then switched virtualenvs.

It turns out that if you use `pyenv` and have pre-commit (or other utilities) installed with a Python it manages, it turns into a pain in the butt, so I ended up installing `pre-commit` via `pipx` with my system python (3.11.5). It turns out there is a [bug in a recent version of flake8](https://stackoverflow.com/questions/77401175/how-to-make-flake8-ignore-syntax-within-strings) that was fixed in 6.1.0.

tl;dr @jendrikseipp I think the fix is just to bump flake8 in `tox.ini`. Do you want a separate PR in the name of a clean changelog or should I just throw it in here?

**Comment by jendrikseipp** [comment]
Thanks for the changes and for looking into the flake8 issue! I'll merge this as is to have a clean history. Would be great if you could send another PR for the flake8 issue.

File: tests/test_config.py (hunk 4 above)
Comment (jendrikseipp) [reply tests/test_config.py]: Good catch!
